### ISSUE TITLE
[BUG] Update & PauseService.isPaused issue

### ISSUE THREAD

**Issue by Acros5542** [AoTTG-2/AoTTG-2#475]
**Note**
See comments for updated information.

**Describe the bug**
This one's pretty simple but may take a few tries to replicate. Basically, horses seem to compile speed while the game is paused with the Escape menu or with /pause in multiplayer. They also seem to lose their trajectory and direct their speed straight down rather than in whatever direction they were travelling before the pause occurred. This can be a bit finnicky to replicate, but it's easiest to do if you have a fair amount of speed on your horse prior to pausing and are in air (post jump) above some sort of nonflat surface. Basically, b-hop your horse up the hills on Outside the Walls, then while you have some speed and are mid-jump above the hills, pause the game, wait a while, and prepare for takeoff upon unpausing.

**To Reproduce**
1. Go to Outside the Walls
2. Get on your horse and build up some speed (easiest by repeatedly jumping)
3. While in midair (but still on horseback) above a non-flat surface, such as the hills on Outside the Walls, pause the game
4. Wait a short period of time (more effective the longer you wait)
5. Unpause

**Expected behavior**
While the game is paused, no sort of horse-speed witchcraft should be occurring.

**Video**
https://youtu.be/qHht-wbJI3M

**Contributor instructions**
1. Try to recreate the bug via the reported reproduction steps. If the bug can be reproduced, remove the investigation label. If the bug cannot be reproduced, please add a comment, attach "On Hold" label and inform the submitter that the bug cannot be reproduced. If there is no reply after 2 weeks, the issue can be closed.
2. Follow the default development team procedures (Create a new branch, apply your fix, create a pull request, submit new builds for testing team in dev squad)
3. Once testing team has confirmed the issue is resolved and the pull request is accepted, the issue can be closed.

**Comment by Lithrun** [comment]
Horses aren't implemented yet. Hard to consider this a bug until #116 has been implemented.

**Comment by Acros5542** [comment]
Reopening as I've discovered this doesn't only affect horses. What I believe is happening is the angle at which movement is occurring updates while paused, so if your horse is in air, normally you would come back down in an arc as your momentum continues to move you forwards, but gravity continually pulls you downwards. However, if you pause the game, the game continues to calculate what angle this movement should be occurring in even though you aren't moving forwards anymore. This means that effectively, when you unpause, all the momentum you had is converted directly downwards, as if you stay paused long enough this calculation for what angle you should be moving in will eventually reach a directly downwards angle.
You can see this by frame-by-framing the horse clips above to see I initially travel downwards, then my force redirects to a different angle since the terrain is not flat.
You can also see how this affects not just horses via this clip of a Punk/Abnormal titan which was running into a tree branch while paused, giving him a slight degree of downwards angular force as he pushed into the angle of the branch. Upon pausing and unpausing a brief while later, all of his force was converted directly downwards, resulting in what is shown in the clip.
https://youtu.be/lvKRpPbQl9s

Tl;dr for math nerds - the direction of the velocity of some entities updates while the game is paused based on the state the entity is in when the game is paused.
